omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java

```diff
@@ -0,0 +1,220 @@
+		
+	}
+	
+	  @Test 
+	  public void getAttachmentBytes_shouldGetFileWhenViewIsNotPassed() throws Exception {
+	 
+		// Setup
+		String fileCaption = "Test file caption";
+		String uuid;
+		String mimeType = "application/octet-stream";
+		String fileExtension = "dat";
+		String fileName = "testFile." + fileExtension;
+		
+		// Upload Test File
+		Patient patient = Context.getPatientService().getPatient(2);
+		MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, mimeType, randomData);
+		
+		uploadRequest.addFile(file);
+		uploadRequest.addParameter("patient", patient.getUuid());
+		uploadRequest.addParameter("fileCaption", fileCaption);
+		
+		SimpleObject uploadResponse = deserialize(handle(uploadRequest));
+		uuid = uploadResponse.get("uuid");
+        
+		HttpServletRequest downloadRequest = newGetRequest(getURI() + "/" + uuid + "/bytes");
+		
+		// Replay
+		MockHttpServletResponse downloadResponse = handle(downloadRequest);
+		byte[] bytesContent = downloadResponse.getContentAsByteArray();
+		
+		// Verify
+		Assert.assertArrayEquals(randomData, bytesContent);
+		Assert.assertEquals(downloadResponse.getContentType(), mimeType);
+		Assert.assertEquals(downloadResponse.getHeader("File-Name"), fileName);
+		Assert.assertEquals(downloadResponse.getHeader("File-Ext"), fileExtension);
+		}
+
+		@Test
+		public void getAttachmentBytes_shouldGetOriginalFileWhenViewIsPassedEqualToUnknown() throws Exception {
+			
+			// Setup
+			String fileCaption = "Test file caption";
+			String uuid;
+			String mimeType = "application/octet-stream";
+			String fileExtension = "dat";
+			String fileName = "testFile." + fileExtension;
+			
+			// Upload Test File
+			Patient patient = Context.getPatientService().getPatient(2);
+			MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+			MockMultipartFile file = new MockMultipartFile("file", fileName, mimeType, randomData);
+			
+			uploadRequest.addFile(file);
+			uploadRequest.addParameter("patient", patient.getUuid());
+			uploadRequest.addParameter("fileCaption", fileCaption);
+			
+			SimpleObject uploadResponse = deserialize(handle(uploadRequest));
+			uuid = uploadResponse.get("uuid");
+			
+			HttpServletRequest downloadRequest = newGetRequest(getURI() + "/" + uuid + "/bytes",new Parameter("view", "unknown"));
+			
+			// Replay
+			MockHttpServletResponse downloadResponse = handle(downloadRequest);
+			byte[] bytesContent = downloadResponse.getContentAsByteArray();
+			
+			// Verify
+			Assert.assertArrayEquals(randomData, bytesContent);
+			Assert.assertEquals(downloadResponse.getContentType(), mimeType);
+			Assert.assertEquals(downloadResponse.getHeader("File-Name"), fileName);
+			Assert.assertEquals(downloadResponse.getHeader("File-Ext"), fileExtension);
+			
+		}
+		@Test
+		public void getAttachmentBytes_shouldGetThumbNailFileWhenViewIsPassedEqualToThumbnail() throws Exception {
```

Comment:
@mks-d I was under the impression that this tests to see the returned bytesContent when we try with the thumbnail view which is zero as compared to the originalview

Reply:
Correct @reagan-meant, I missed that. I will review this again.

Reply:
@mks-d I will be waiting for ur feedback